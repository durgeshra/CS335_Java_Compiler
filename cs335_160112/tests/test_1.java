// Taken from https://www.codechef.com/viewsolution/2988299
import java.io.IOException;
import java.io.InputStream;
import java.util.Arrays;
import java.util.InputMismatchException;
 
public class Main {
 
	/**
	 * @param args
	 * @throws IOException 
	 * @throws NumberFormatException 
	 */
	public static void main(String[] args) throws NumberFormatException, IOException {
		InputReader in=new InputReader(System.in);
		//BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
		int t=in.readInt();
		while(t-->0){
			
			int n=in.readInt();
			int k=in.readInt();
			int vect[][]=new int[n][k];
			int a[]=new int[k];
			int i=0;
			while(i<n){
				
				int j=0;
				while(j<k){
					vect[i][j]=in.readInt();
					j++;
				}
				i++;
			}
			int j=0;
			
			
			while(j<k){
				a[j]=in.readInt();
				j++;
			}			
			
			i=0;
			
			boolean whoisvalid[]=new boolean[n];
			/*
			int looptime=1000;
			if(n*k>1000 && n*k<5000) looptime=500;
			else if(n*k>10000) looptime=50;
			
			*/
			int looptime =0;
			int ans[]=new int[1000];
			int q=0;
			while(looptime<1000000){
				
				i=0;
				while(i<n){
					j=0;
					int temp=0;
					while(j<k){
						if(vect[i][j]>a[j] || vect[i][j]==-1) { temp=1;break;}
						j++;
					}
					if(temp==0) {
						whoisvalid[i]=true;
					}else whoisvalid[i]=false;
					i++;
				}
				i=0;
				int temp=0;	
				long sum=0;long min=0L,index=0;
				while(i<n){
					sum=0;
					if(whoisvalid[i]==true) {
						int kk=0;
						while(kk<k){
							sum+=a[kk]-vect[i][kk];
							kk++;
						}
						if(min<sum){
							min=sum;
							index=i;
						}
						temp=1;
 
					}
					i++;
				}
				if(temp==0){
					break;
				}
				else {
					//System.out.println(1);
					int kk=0;
					
					
						while(kk<k){
							a[kk]=a[kk]-vect[(int)index][kk];
							vect[(int)index][kk]=-1;
							kk++;
						}
					
					ans[q]=(int)index+1;
					q++;
				//	System.out.println(index+1); 
					whoisvalid[(int) index]=false;
					if(looptime*n*k*100>=100000000) break;
					looptime++;
				}
			}
			if(q==0){
				System.out.println(0);
			}else {
				Arrays.sort(ans,0,q);
				int kk=0;
				System.out.println(q);
				while(kk<q){
					System.out.print(ans[kk]+" ");
					kk++;
				}
				System.out.println();
			}
			
		}
	}
 
} 
 
 
class InputReader {
	 
 
	
	private InputStream stream;
    private byte[] buf = new byte[1024];
    private int curChar;
    private int numChars;
 
    public InputReader(InputStream stream) {
        this.stream = stream;
    }
 
    public int read() {
        if (numChars == -1)
            throw new InputMismatchException();
        if (curChar >= numChars) {
            curChar = 0;
            try {
                numChars = stream.read(buf);
            } catch (IOException e) {
                throw new InputMismatchException();
            }
            if (numChars <= 0)
                return -1;
        }
        return buf[curChar++];
    }
 
    public int readInt() {
        int c = read();
        while (isSpaceChar(c))
            c = read();
        int sgn = 1;
        if (c == '-') {
            sgn = -1;
            c = read();
        }
        int res = 0;
        do {
            if (c < '0' || c > '9')
                throw new InputMismatchException();
            res *= 10;
            res += c - '0';
            c = read();
        } while (!isSpaceChar(c));
        return res * sgn;
    }
 
    public String readString() {
        int c = read();
        while (isSpaceChar(c))
            c = read();
        StringBuffer res = new StringBuffer();
        do {
            res.appendCodePoint(c);
            c = read();
        } while (!isSpaceChar(c));
        return res.toString();
    }
 
    public static boolean isSpaceChar(int c) {
        return c == ' ' || c == '\n' || c == '\r' || c == '\t' || c == -1;
    }
}  
