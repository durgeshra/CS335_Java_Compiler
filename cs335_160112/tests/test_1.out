! Operator 2
" " Literal 1
' ' Literal 1
'-' Literal 1
'0' Literal 2
'9' Literal 1
'\n' Literal 1
'\r' Literal 1
'\t' Literal 1
( Separator 67
) Separator 67
* Operator 4
*= Operator 1
+ Operator 2
++ Operator 12
+= Operator 2
, Separator 4
- Operator 9
-- Operator 1
. Separator 27
0 Literal 26
0L Literal 1
1 Literal 10
10 Literal 1
100 Literal 1
1000 Literal 1
1000000 Literal 1
100000000 Literal 1
1024 Literal 1
; Separator 92
< Operator 12
<= Operator 1
= Operator 53
== Operator 12
> Operator 3
>= Operator 2
Arrays Identifier 2
IOException Identifier 3
InputMismatchException Identifier 4
InputReader Identifier 4
InputStream Identifier 3
Main Identifier 1
NumberFormatException Identifier 1
String Identifier 2
StringBuffer Identifier 2
System Identifier 5
[ Separator 37
] Separator 37
a Identifier 6
ans Identifier 4
appendCodePoint Identifier 1
args Identifier 1
boolean Keyword 3
break Keyword 3
buf Identifier 3
byte Keyword 2
c Identifier 22
catch Keyword 1
class Keyword 2
curChar Identifier 4
do Keyword 2
e Identifier 1
else Keyword 3
false Literal 2
i Identifier 18
if Keyword 12
import Keyword 4
in Identifier 7
index Identifier 6
int Keyword 32
io Identifier 2
isSpaceChar Identifier 5
j Identifier 14
java Identifier 4
k Identifier 9
kk Identifier 16
long Keyword 2
looptime Identifier 4
main Identifier 1
min Identifier 3
n Identifier 7
new Keyword 10
numChars Identifier 5
out Identifier 4
print Identifier 1
println Identifier 3
private Keyword 4
public Keyword 7
q Identifier 7
read Identifier 9
readInt Identifier 6
readString Identifier 1
res Identifier 7
return Keyword 5
sgn Identifier 3
sort Identifier 1
static Keyword 2
stream Identifier 5
sum Identifier 5
t Identifier 2
temp Identifier 6
this Keyword 1
throw Keyword 3
throws Keyword 1
toString Identifier 1
true Literal 2
try Keyword 1
util Identifier 2
vect Identifier 7
void Keyword 1
while Keyword 15
whoisvalid Identifier 5
{ Separator 33
|| Operator 6
} Separator 33
